Xamarin.Forms.Build.Tasks/XamlCTask.cs

```diff
@@ -147,24 +152,29 @@ public override bool Execute(out IList<Exception> thrownExceptions)
 												  (string)xamlFilePathAttr.ConstructorArguments[0].Value :
 												  resource.Name;
 
-						var initCompRuntime = typeDef.Methods.FirstOrDefault(md => md.Name == "__InitComponentRuntime");
-						if (initCompRuntime != null)
-							LoggingHelper.LogMessage(Low, $"{new string(' ', 6)}__InitComponentRuntime already exists... not creating");
-						else {
-							LoggingHelper.LogMessage(Low, $"{new string(' ', 6)}Creating empty {typeDef.Name}.__InitComponentRuntime");
-							initCompRuntime = new MethodDefinition("__InitComponentRuntime", initComp.Attributes, initComp.ReturnType);
-							initCompRuntime.Body.InitLocals = true;
-							LoggingHelper.LogMessage(Low, $"{new string(' ', 8)}done.");
-							LoggingHelper.LogMessage(Low, $"{new string(' ', 6)}Copying body of InitializeComponent to __InitComponentRuntime");
-							initCompRuntime.Body = new MethodBody(initCompRuntime);
-							var iCRIl = initCompRuntime.Body.GetILProcessor();
-							foreach (var instr in initComp.Body.Instructions)
-								iCRIl.Append(instr);
-							initComp.Body.Instructions.Clear();
-							initComp.Body.GetILProcessor().Emit(OpCodes.Ret);
-							initComp.Body.InitLocals = true;
-							typeDef.Methods.Add(initCompRuntime);
-							LoggingHelper.LogMessage(Low, $"{new string(' ', 8)}done.");
+						MethodDefinition initCompRuntime = null;
+						if (!ReadOnly)
```

Comment:
I would disable the writing at the actual place we write the assembly back
```
assemblyDefinition.Write(new WriterParameters {
						WriteSymbols = debug,
					});
```

what you're removing here is the duplication of the XamlG generated InitializeComponent (used by the previewer).

we still modify the assembly in TryCoreCompile, while setting the mvid, or when pruning resources

Reply:
I removed this block because there didn't seem to be a reason to call Mono.Cecil to modify things, if it is not going to write?

I was just trying to skip a block of code that seemed to not validate anything, is that fine?

Reply:
it is fine, but I don't think this waste any time. we still have all the other code generating tons more of IL than this